pydantic/main.py

```diff
@@ -890,53 +893,64 @@ def __getattr__(self, item: str) -> Any:
                         raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
 
         def __setattr__(self, name: str, value: Any) -> None:
-            if name in self.__class_vars__:
+            if (fast_memo_handler := self.__pydantic_setattr_handlers__.get(name)) is not None:
+                fast_memo_handler(self, value)
```

Comment:
I think the word `fast` in the name of this variable is carrying some weight in terms of documenting what is going on here — it doesn't seem to be indicated explicitly that the reason for memoizing the attribute handler is purely for performance and isn't otherwise logically required. I mean, I know we have the docstring of `__pydantic_setattr_handlers__`, but that is far away from the implementing code and still doesn't make it clear that the current logic _only_ makes use of the memoization for performance reasons (and isn't otherwise dependent on the memoization), which may be useful context if we refactor this in the future.

I don't really mind the word fast in the name, but I think either way we should add a comment here reiterating that the memoization is purely for performance reasons. And if we do, the word fast probably becomes less useful.

Reply:
Done, simplified the name, it wasnt so pretty yet :)